src/EachPromise.php

```diff
@@ -10,6 +10,8 @@ class EachPromise implements PromisorInterface
 {
     private $pending = [];
 
+    private $nextPendingIndex = 0;
```

Comment:
seems fine to do it this way. But I suggest to reset the counter in the clear function in https://github.com/guzzle/promises/blob/0e3f22db085e896f13c0a4e6cf3d916f65de2358/src/EachPromise.php#L123